File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 0:
            return []

        mp = {}
        
        for i, num in enumerate(nums):
            diff = target - num
            if diff not in mp:
                mp[num] = i
            else:
                return [mp[diff], i]

        return []
        

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-3.cpp
class Solution {
public:
    bool mergeTriplets(vector<vector<int>>& triplets, vector<int>& target) {
        int i = 0, j = 1, k = 2, n = triplets.size();
        int aI = 0, bI = 0, cI = 0;

        for (int x = 0; x < n; ++x) {
            if (triplets[x][i] > target[i] ||triplets[x][j] > target[j] || triplets[x][k] > target[k]) continue;
            if (triplets[x][i] <= target[i] && triplets[x][j] <= target[j] && triplets[x][k] <= target[k]) {
                aI = max(aI, triplets[x][i]);
                bI = max(bI, triplets[x][j]);
                cI = max(cI, triplets[x][k]);
            }
        }
        return aI == target[i] && bI == target[j] && cI == target[k];
    }
};
